stonesoup/detector/beamformers.py

```diff
@@ -0,0 +1,544 @@
+
+                        # (NOTE cont'd) ... these:
+                        # # steering vector
+                        # v = np.zeros(N, dtype=np.complex)
+                        #
+                        # # directional unit vector
+                        # a = np.zeros(3)
+                        #
+                        # # convert from spherical polar coordinates to cartesian
+                        # a[0] = math.cos(theta) * math.sin(phi)
+                        # a[1] = math.sin(theta) * math.sin(phi)
+                        # a[2] = math.cos(phi)
+                        # a = a / math.sqrt(np.sum(a * a))
+                        # for n in range(0, N):
+                        #     phase = np.sum(a * np.transpose(z[n,])) / c
+                        #     v[n] = math.cos(phase) - math.sin(phase) * 1j
+
+                        F = 1 / ((L - N) * np.conj(v).T @ R_inv @ v)
+                        if F > maxF:
+                            maxF = F
+                            maxtheta = theta
+                            maxphi = phi
+
+                # Defining a detection
+                state_vector = StateVector([maxtheta, maxphi])  # [Azimuth, Elevation]
+                covar = CovarianceMatrix(np.array([[1, 0], [0, 1]]))
+                measurement_model = LinearGaussian(ndim_state=4, mapping=[0, 2],
+                                                   noise_covar=covar)
+                current_time = current_time + timedelta(milliseconds=1000*L/self.fs)
+                detection = Detection(state_vector, timestamp=current_time,
+                                      measurement_model=measurement_model)
+
+                yield current_time, {detection}
+
+
+class RJMCMCBeamformer(Base, BufferedGenerator):
+    """A parameter estimation algorithm for a sensor array measuring passive signals. Given the
+    input signals from the array, the algorithm uses reversible-jump Markov chain Monte Carlo [1]
+    to sample from the posterior probability for a model where the number of targets and directions
+    of arrival are the unknown parameters. The algorithm is based on the work in [2] and [3].
+
+    [1] P. J. Green, Reversible jump Markov chain Monte Carlo computation and Bayesian
+    model determination, Biometrika 82(4):711-732 (1995)
+    [2] C. Andrieu and A. Doucet, Joint Bayesian Model Selection and Estimation of
+    Noisy Sinusoids via Reversible Jump MCMC, IEEE Trans. Signal Process. 47(10):2667-2676 (1999)
+    [3] C. Andrieu, N. de Freitas and A. Doucet, Robust Full Bayesian Learning for Radial Basis
+    Networks, Neural Computation 13:2359–2407 (2001)
+
+    """
+    path: str = Property(doc='The path to the csv file, containing the raw data')
+    fs: float = Property(doc='Sampling frequency (Hz)')
+    omega: float = Property(doc='Signal frequency (Hz)')
+    sensor_loc: list = Property(doc='Cartesian coordinates of the sensors in the format\
```

Comment:
For `sensor_loc`, having a list of state vectors might be better `Sequence[StateVector]` (as mentioned by SH... I think)